src/commands/init/ai-rules.ts

```diff
@@ -0,0 +1,447 @@
+  if (process.env.AI_CONTEXT_SKIP_DETECTION === 'true') {
+    return null
+  }
+
+  const ppid = process.ppid
+  let result: Awaited<ReturnType<typeof getCommandAndParentPID>>
+  try {
+    result = await getCommandAndParentPID(ppid)
+    while (result.parentPID !== 1 && !result.consumerKey) {
+      result = await getCommandAndParentPID(result.parentPID)
+    }
+  } catch {
+    // Process detection failed
+    return null
+  }
+
+  if (result.consumerKey) {
+    const consumers = await getContextConsumers(version)
+    const contextConsumer = consumers.find((consumer) => consumer.key === result.consumerKey)
+    if (contextConsumer) {
+      return contextConsumer
+    }
+  }
+
+  return null
+}
+
+// Generate MCP configuration for the detected IDE
+const generateMcpConfig = (ide: ConsumerConfig): string => {
+  const configs: Record<string, string> = {
+    vscode: JSON.stringify(
+      {
+        servers: {
+          netlify: {
+            type: 'stdio',
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    ),
+    cursor: JSON.stringify(
+      {
+        mcpServers: {
+          netlify: {
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    ),
+    windsurf: JSON.stringify(
+      {
+        mcpServers: {
+          netlify: {
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    ),
+  }
+
+  return (
+    configs[ide.key] ||
+    JSON.stringify(
+      {
+        mcpServers: {
+          netlify: {
+            command: 'npx',
+            args: ['-y', '@netlify/mcp'],
+          },
+        },
+      },
+      null,
+      2,
+    )
+  )
+}
+
+// VS Code specific MCP configuration
+const configureMcpForVSCode = async (config: string, projectPath: string): Promise<void> => {
+  const configPath = resolve(projectPath, '.vscode', 'mcp.json')
+
+  try {
+    // Create .vscode directory if it doesn't exist
+    await fs.mkdir(resolve(projectPath, '.vscode'), { recursive: true })
+
+    // Write or update mcp.json
+    let existingConfig: Record<string, unknown> = {}
+    try {
+      const existingContent = await fs.readFile(configPath, 'utf-8')
+      existingConfig = JSON.parse(existingContent) as Record<string, unknown>
+    } catch {
+      // File doesn't exist or is invalid JSON
+    }
+
+    const mcpConfig = JSON.parse(config) as Record<string, unknown>
+    const updatedConfig = { ...existingConfig, ...mcpConfig }
+
+    await fs.writeFile(configPath, JSON.stringify(updatedConfig, null, 2), 'utf-8')
+    log(`${chalk.green('✅')} VS Code MCP configuration saved to ${chalk.cyan('.vscode/mcp.json')}`)
+  } catch (error) {
+    throw new Error(`Failed to configure VS Code MCP: ${error instanceof Error ? error.message : 'Unknown error'}`)
```

Comment:
When will `error` not be an instance of `Error`?

Reply:
it's always unknown, but here yeah it will be Error, it's more following code pattern across repo